sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -463,7 +463,7 @@ object DottyPlugin extends AutoPlugin {
         Def.task { tastyFiles.value }
       else Def.task { originalSources }
     }.value,
-    scalacOptions ++= {
+    scalacOptions := {
```

Comment:
I am not convinced we should do that. We share some options with the compiler and people are setting them using scalacOptions setting that we inherit

Reply:
These are options for scaladoc. The default `scalacOptions` are from `val thisBuildSettings`:
```
    Seq(
      "-feature",
      "-deprecation",
      "-unchecked",
      "-Xfatal-warnings",
      "-encoding", "UTF8",
      "-language:existentials,higherKinds,implicitConversions,postfixOps"
    )
```
which are not even handled nor recognized by scaladoc. They are always added because we default them with `inThisBuild` util function. The other approach to the problem would be breaking the `thisBuildSettings` into `commonThisBuildSettings` and `scalaThisBuildSettings` and `scaladocThisBuildSettings`, but I decided to just ignore the default options since they are not sufficient in our context